### ISSUE TITLE
environment/initializers not loaded

### ISSUE THREAD

**Issue by nickjacob** [ad2games/rails_migrate_mutex#1]
Not sure if this is something wrong in my setup, but this wasn't working for me since I need to set the `RedisClassy`/`Redis` instance based on an initializer (as mentioned in the readme).

Got it to work by overriding task with my own, where I depend on environment:

``` ruby
namespace :db do
  namespace :migrate do
    desc 'Runs db:migrate in a redis-backed mutex'
    # this change let me load initializers
    task :mutex => :environment do
      TIMEOUT = 5 * 60 * 60

      RedisClassy.redis ||= Redis.current

      RedisMutex.with_lock('db-migrate-mutex', block: TIMEOUT, expire: TIMEOUT) do
        Rake::Task['db:migrate'].invoke
      end
    end
  end
end
```

Am I missing something or should I open a PR with this change?

**Comment by dziemba** [comment]
Hi, thanks for reporting this. I think you're right, we need to depend on the `environment` task to properly load the initializers. I just never noticed this because our apps don't need a custom redis config. 

Feel free to open a PR!

**Comment by mygithubnamerules** [comment]
I just spent an hour or two tracking this down because I didn't think to look here. Here is a hack which will allow future travelers to get around this issue until it's fixed.... 


#This assumes that you've set up the redis_classy initializer. 
task :with_redis => :environment do
 puts "enhancing with redis"
end

Rake::Task["db:migrate:mutex"].enhance [:with_redis]

**Comment by dziemba** [comment]
Thanks for all your input.

It's now fixed, see https://github.com/ad2games/rails_migrate_mutex/commit/95cddaea0390b1d85014e512632b829a9c54880c

It's also released as version `1.2.0`.
